ZoneEndPoint2D[] points
Area area   ##define where this special area belong to
Type type # 0=normal 1=single 2=both